src/Mapbender/CoreBundle/Resources/public/mapbender.element.layertree.tree.js

```diff
@@ -170,100 +170,37 @@
-                if ($afterItem.hasClass('themeContainer')) {
-                    var $afterSEl = $afterItem.find('ul.layers:first li[data-type="root"]:first');
-                    var bid = $afterSEl.attr('data-sourceid');
-                    after = {
-                        sourceIdx: {
-                            id: bid
-                        },
-                        layerIdx: {
-                            id: $afterSEl.attr("data-id")
-                        }
-                    };
-                } else {
-                    var bid = $afterItem.attr('data-sourceid');
-                    bid = bid ? bid : $afterItem.parents('li[data-sourceid]:first').attr('data-sourceid');
-                    after = {
-                        sourceIdx: {
-                            id: bid
-                        },
-                        layerIdx: {
-                            id: $afterItem.attr("data-id")
-                        }
-                    };
-                }
-            }
-            var id = $toMoveItem.attr('data-sourceid');
-            id = id ? id : $toMoveItem.parents('li[data-sourceid]:first').attr('data-sourceid');
-            toMove = {
-                sourceIdx: {
-                    id: id
-                }
-            };
-            if ($toMoveItem.attr("data-type") !== this.consts.root) {
-                toMove['layerIdx'] = {
-                    id: $toMoveItem.attr("data-id")
-                };
-            }
-            this.model.changeSource({
-                change: {
-                    move: {
-                        tomove: toMove,
-                        before: after,
-                        after: before
-                    }
-                }
-            });
-        },
-        _sortTheme: function($theme) {
-            var self = this;
-            var $beforeEl = $theme.prev().length !== 0 ? $($theme.prev()[0]) : null;
-            var $afterEl = $theme.next().length !== 0 ? $($theme.next()[0]) : null;
-            var $last = null;
-            $('ul.layers:first li[data-type="root"]', $theme).each(function(idx, item) {
-                var $item = $(item);
-                if ($last) {
-                    $beforeEl = $last;
+        /**
+         * Applies the new (going by DOM) layer order inside a source.
+         *
+         * @param $sourceContainer
+         * @private
+         */
+        _updateSource($sourceContainer) {
+            // this will capture the "configurationish" layer ids (e.g. "1_0_4_1") from
+            // all layers in the source container in DOM order
+            var sourceId = $sourceContainer.attr('data-sourceid');
+            var layerIdOrder = [];
+            $('li.leave[data-type="simple"]', $sourceContainer).each(function() {
+                var $t = $(this);
+                var layerId = $t.attr('data-id');
+                if (layerId !== undefined) {
+                    layerIdOrder.push(layerId.toString());
```

Comment:
Please stick to one way of int to string cast. Further in the code ```""+int``` is used which is the faster way to do.

Reply:
[Done?](https://github.com/mapbender/mapbender/pull/852/commits/c992508c9fca4dd591d0652255f0d5e4b334b7fc)